src/components/pages/rules/flyouts/ruleEditor/ruleEditor.js

```diff
@@ -95,36 +99,35 @@ export class RuleEditor extends LinkedComponent {
       this.ruleNameLink,
       this.deviceGroupLink,
       this.conditionsLink,
+      this.timePeriodLink,
+      this.calculationLink
     ].every(link => !link.error);
   }
 
   apply = (event) => {
     event.preventDefault();
     const { formData } = this.state;
     const { onClose, insertRule, updateRule } = this.props;
-    console.log('TODO: Handle the form submission');
+    if (formData.calculation === calculations[1]) this.setState({ formData: { ...this.state.formData, timePeriod: '' } })
     if (this.formIsValid()) {
-      this.setState({ isPending: true });
       if (this.subscription) this.subscription.unsubscribe();
       if (this.props.rule) { // If rule object exist then update the existing rule
         this.subscription = TelemetryService.updateRule(this.props.rule.id, toNewRuleRequestModel(formData))
           .subscribe(
             (updatedRule) => {
               updateRule(updatedRule);
-              this.setState({ isPending: false });
               onClose();
             },
-            error => this.setState({ error, isPending: false })
+            error => this.setState({ error })
           );
       } else { // If rule object doesn't exist then create a new rule
         this.subscription = TelemetryService.createRule(toNewRuleRequestModel(formData))
           .subscribe(
             (createdRule) => {
               insertRule(createdRule);
-              this.setState({ isPending: false });
               onClose();
             },
-            error => this.setState({ error, isPending: false })
+            error => this.setState({ error })
```

Comment:
You'll still need to know the pending status to alert the user that their is an action happening. Why are you removing it?

Reply:
That was not in the redline or in previous version

Reply:
The previous version is wrong. If it isn't in the redline it probably should be. You should ask about it.